books/kestrel/c/transformation/split-fn.lisp

```diff
@@ -0,0 +1,457 @@
+; C Library
+;
+; Copyright (C) 2024 Kestrel Institute (http://www.kestrel.edu)
+;
+; License: A 3-clause BSD license. See the LICENSE file distributed with ACL2.
+;
+; Author: Grant Jurgensen (grant@kestrel.edu)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(in-package "C2C")
+
+(include-book "std/util/bstar" :dir :system)
+(include-book "std/util/define" :dir :system)
+(include-book "std/util/defrule" :dir :system)
+(include-book "xdoc/defxdoc-plus" :dir :system)
+(include-book "xdoc/constructors" :dir :system)
+
+(include-book "centaur/fty/deftypes" :dir :system)
+(include-book "kestrel/std/util/defirrelevant" :dir :system)
+(include-book "kestrel/std/util/error-value-tuples" :dir :system)
+
+(include-book "../syntax/abstract-syntax-operations")
+(include-book "deftrans")
+(include-book "utilities/free-vars")
+
+(local (include-book "kestrel/built-ins/disable" :dir :system))
+(local (acl2::disable-most-builtin-logic-defuns))
+(local (acl2::disable-builtin-rewrite-rules-for-defaults))
+(set-induction-depth-limit 0)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defxdoc+ split-fn
+  :parents (transformation-tools)
+  :short "A C-to-C transformation to split a function in two."
+  :long
+  (xdoc::topstring
+    (xdoc::p
+      "This transformation takes the identifier of the function it is to split,
+       the name of the new function to be generated, and the location of the
+       split, represented as a natural number corresponding to the number of
+       statement block items in the function body before the split.")
```

Comment:
Is this the number of block items that are statements, or just the number of block items?